src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -190,4 +229,17 @@ public static void generatePemX509Certificate(ByteStringBuilder target, X509Cert
             throw log.certificateParseError(e);
         }
     }
+
+    /**
+     * Generate PEM content containing a {@link PublicKey}.
+     *
+     * @param target the target byte string builder (must not be {@code null})
+     * @param publicKey the {@link PublicKey} (must not be {@code null})
+     */
+    public static void generatePemPublicKey(ByteStringBuilder target, PublicKey publicKey) {
+        Assert.checkNotNullParam("target", target);
+        Assert.checkNotNullParam("publicKey", publicKey);
+
+        generatePemContent(target, "PUBLIC KEY", ByteIterator.ofBytes(publicKey.getEncoded()));
```

Comment:
This would break if the key doesn't use the X.509 representation, right?  In this case we should use a KeyFactory for the key to get an X.509 keyspec and then persist that, I think.